pyam/utils.py

```diff
@@ -262,12 +262,18 @@ def _intuit_column_groups(df, index, include_index=False):
         existing_cols = existing_cols.union(df.columns)
 
     # check that there is no column in the timeseries data with reserved names
+    if None in existing_cols:
+        raise ValueError("Unnamed column in `data`: None")
```

Comment:
This could also be part of the `ILLEGAL_COLS`, i guess, but maybe the message is just clearer this way.

Reply:
Right, the reason why this is not part `ILLEGAL_COLS` is that we have an easy solution for string-column-names, which is part of the error message - but this doesn't work (easily) for `None`.